src/delegation/store/traits.rs

```diff
@@ -1,25 +1,39 @@
 use crate::{
+    ability::arguments::Named,
+    crypto::signature::Envelope,
     crypto::varsig,
+    delegation::payload::Payload,
     delegation::{policy::Predicate, Delegation},
     did::Did,
 };
+use libipld_core::codec::Encode;
+use libipld_core::ipld::Ipld;
 use libipld_core::{cid::Cid, codec::Codec};
 use nonempty::NonEmpty;
 use std::{fmt::Debug, sync::Arc};
 use web_time::SystemTime;
 
-pub trait Store<DID: Did, V: varsig::Header<Enc>, Enc: Codec + TryFrom<u64> + Into<u64>> {
+pub trait Store<DID: Did + Clone, V: varsig::Header<C> + Clone, C: Codec + TryFrom<u64> + Into<u64>>
+where
+    Ipld: Encode<C>,
+    Payload<DID>: TryFrom<Named<Ipld>>,
+    Named<Ipld>: From<Payload<DID>>,
```

Comment:
This combination of constraints comes up repeatedly a bit.
Perhaps worth keeping in the back of our minds whether we should turn this into a trait that combines all these constraints? Personally I'd also appreciate getting away from `From` and `TryFrom` for parsing. IMO they can make code & compiler errors less readable.
Not in this PR though! Let's just keep it in mind.